### PR TITLE
LPD-25679 Add onSetData prop to RichTextLocalized component

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://liferay.atlassian.net/browse/LPD-25679

### CODE REVIEW

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/RichTextLocalized.tsx [hunk 1]
```diff
@@ -18,8 +18,59 @@ import React, {useEffect, useRef, useState} from 'react';
...
+
 const defaultLanguageId = Liferay.ThemeDisplay.getDefaultLanguageId();
 
+const ALERT_REGEX = /alert\((.*?)\)/;
+const INNER_HTML_REGEX = /innerHTML\s*=\s*.*?/;
+const PHP_CODE_REGEX = /<\?[\s\S]*?\?>/g;
+const ASP_CODE_REGEX = /<%[\s\S]*?%>/g;
+const ASP_NET_CODE_REGEX = /(<asp:[^]+>[\s|\S]*?<\/asp:[^]+>)|(<asp:[^]+\/>)/gi;
+const HTML_TAG_WITH_ON_ATTRIBUTE_REGEX = /<[^>]+?(\s+\bon\w+=(?:'[^']*'|"[^"]*"|[^'"\s>]+))*\s*\/?>/gi;
+const ON_ATTRIBUTE_REGEX = /(\s+\bon\w+=(?:'[^']*'|"[^"]*"|[^'"\s>]+))/gi;
```
Comment (igor-franca) [inline modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/RichTextLocalized.tsx:58]: Hey @danvb112, could you sort these constants?

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/RichTextLocalized.tsx [hunk 2]
```diff
@@ -18,8 +18,59 @@ import React, {useEffect, useRef, useState} from 'react';
...
+	translations: LocalizedValue<string>;
+}
+
+interface OnSetDataEvent {
+	data: {
+		dataValue: string;
+	};
+	editor: CKEDITOR.editor;
+}
+
```
Comment (igor-franca) [inline modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/RichTextLocalized.tsx:49]: Hey @danvb112, could you sort these interfaces?

**Review by igor-franca — CHANGES_REQUESTED** [review]
Hey @danvb112, just one detail 😄

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:forward:force

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:forward:force

**Comment by danvb112** [comment]
CI failures are unrelated, manually sending to Brian

**Comment by danvb112** [comment]
Sent to: https://github.com/brianchandotcom/liferay-portal/pull/150358

**Pull request closed without merging**